File: modules/AcroStack.BackgroundJobs/Dtos/BackgroundJobDto.cs
using System;
using Volo.Abp.BackgroundJobs;

namespace AcroStack.BackgroundJobs;

public class BackgroundJobDto
{
    public Guid Id { get; set; }
    public string? JobName { get; set; }
    public string? JobArgs { get; set; }
    public short TryCount { get; set; }
    public DateTime CreationTime { get; set; }
    public DateTime NextTryTime { get; set; }
    public DateTime? LastTryTime { get; set; }
    public bool IsAbandoned { get; set; }

    /// <summary>任务完成时间（未完成为 null）。</summary>
    public DateTime? CompletionTime { get; set; }

    /// <summary>产生任务的应用名称。</summary>
    public string? ApplicationName { get; set; }

    /// <summary>任务优先级（值越大越优先）。</summary>
    public BackgroundJobPriority? Priority { get; set; }
}

File: react/src/api/models/acroStack/backgroundJobs/BackgroundJobDto.ts
/* oxlint-disable */

import type { VoloAbpBackgroundJobsBackgroundJobPriority } from "../../volo/abp/backgroundJobs/BackgroundJobPriority";

export type AcroStackBackgroundJobsBackgroundJobDto = {
  /**
   * @description
   * Format: `uuid`
   * @type string | undefined
   */
  id?: string;
  jobName?: string | null;
  jobArgs?: string | null;
  /**
   * @description
   * Format: `int32`
   * @type integer | undefined
   */
  tryCount?: number;
  /**
   * @description
   * Format: `date-time`
   * @type string | undefined
   */
  creationTime?: string;
  /**
   * @description
   * Format: `date-time`
   * @type string | undefined
   */
  nextTryTime?: string;
  /**
   * @description
   * Format: `date-time`
   * @type string | undefined
   */
  lastTryTime?: string | null;
  isAbandoned?: boolean;
  /**
   * @description 任务完成时间（未完成为 null）。
   *
   * Format: `date-time`
   * @type string | undefined
   */
  completionTime?: string | null;
  /**
   * @description 产生任务的应用名称。
   * @type string | undefined
   */
  applicationName?: string | null;
  /**
   * @description
   * Format: `int32`
   * @type integer | undefined
   */
  priority?: VoloAbpBackgroundJobsBackgroundJobPriority;
};
